draft-ietf-ppm-dap.md

```diff
@@ -1371,11 +1345,53 @@ following generic checks.
    the same. If so, then the Aggregator MUST mark the input share as invalid
    with error "unrecognized_message".
 
-1. Finally, if an Aggregator cannot determine if an input share is valid, it
-   MUST mark the input share as invalid with error `report_dropped`. This
-   situation arises if, for example, the Aggregator has evicted from long-term
-   storage the state required to perform the check. (See
-   {{reducing-storage-requirements}} for details.)
+1. If an Aggregator cannot determine if an input share is valid, it
+   MUST mark the input share as invalid with error `report_dropped`. For
+   example, if the Aggregator has evicted the state required to perform the
+   check from long-term storage. (See {{reducing-storage-requirements}} for
+   details.)
+
+1. Some VDAFs allow a batch to be collected multiple times; the aggregators must
+   check that the exact same set of reports is being aggregated each time.
+   A report may not be aggregated into a batch unless it was included in all
+   previous collections for the batch. (Checking that no batches are omitted
+   from a batch is covered in {{batch-validation}}.)
+   If this check fails, the input share MUST be marked as invalid with the error
+   `batch_collected`. This prevents collectors from learning anything about
+   small numbers of reports that are uploaded between two collections of a
+   batch.
+   The helper must consider a batch as collected when it has sent the
+   `AggregateShare` to the leader.
+
+The following additional checks may apply depending on the VDAF and query type.
```

Comment:
I would say the agg param validity scope check should be made for all VDAFs. Where we landed in 04 is to require that each VDAF specify a method for checking it: https://stackoverflow.com/questions/868568/what-do-the-terms-cpu-bound-and-i-o-bound-mean